File: J_se_les12/HW/Core/Presenter/PresenterV1.java
package Java.J_se_les12.HW.Core.Presenter;

import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleInput;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutErrors;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutSuc;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutput;
import Java.J_se_les12.HW.Core.Logic.Interfaces.ILFactory;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IMainMenu;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public class PresenterV1 extends PresenterBase {
    private String path;
    Phonebook phonebook;
    IConsoleOutErrors errors;
    IConsoleOutput output;
    IConsoleOutSuc outSuc;
    IConsoleInput consoleInput;
    IMainMenu menu;

    public PresenterV1(String path, Phonebook phonebook, ILFactory factory, IConsoleOutErrors errors,
            IConsoleOutput output, IConsoleOutSuc outSuc, IConsoleInput consoleInput, IMainMenu menu) {
        super(factory);
        this.path = path;
        this.phonebook = phonebook;
        this.errors = errors;
        this.output = output;
        this.outSuc = outSuc;
        this.consoleInput = consoleInput;
        this.menu = menu;
    }

    @Override
    public void doThis() {
        String choose = "";
        while (!choose.equals("0")) {
            menu.showMainMenu();
            choose = consoleInput.getCommand();
            factory.makeOperation(path, choose, phonebook, errors, output, outSuc, consoleInput);
        }

    }

}

File: J_se_les13/HW/Core/Presenter/PresenterV1.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
namespace HW.Core.Presenter
{
    public class PresenterV1 : PresenterBase
    {
        private string Path { get; set; }
        PhonebookV1 Phonebook { get; set; }
        IConsoleOutError Error { get; set; }
        IConsoleOutput Output { get; set; }
        IConsoleOutSuc OutSuc { get; set; }
        IConsoleInput Input { get; set; }
        IMainMenu Menu { get; set; }
        public PresenterV1(string path, PhonebookV1 phonebook, ILFactory factory, IConsoleOutError error,
            IConsoleOutput output, IConsoleOutSuc outSuc, IConsoleInput input, IMainMenu menu)
        : base(factory)
        {
            Path = path;
            Phonebook = phonebook;
            Factory = factory;
            Error = error;
            Output = output;
            OutSuc = outSuc;
            Input = input;
            Menu = menu;
        }
        public override void DoThis()
        {
            string choose = "";
            while (choose != "0")
            {
                Menu.ShowMenu();
                choose = Input.GetCommand();
                Factory.MakeOperation(Path,choose, Phonebook, Error, Output, OutSuc, Input);
            }
        }

    }
}
